tests/tls_e2e/client_enc/client.cpp

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/edger8r/enclave.h>
+#include <openenclave/enclave.h>
+//#define OE_NEED_STDC_NAMES
+#include <openenclave/corelibc/sys/socket.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/tests.h>
+
+#include <openenclave/corelibc/arpa/inet.h>
+#include <openenclave/corelibc/netdb.h>
+#include <openenclave/corelibc/netinet/in.h>
+#include <openenclave/internal/posix/device.h>
+
+//#include <bits/stdfile.h> // For stderr & FILE
+#include <errno.h> // For errno & error defs
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// clang-format off
+#include <mbedtls/pk.h>
+#include <mbedtls/rsa.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/certs.h>
+#include <mbedtls/x509.h>
+#include <mbedtls/ssl.h>
+#include <mbedtls/net_sockets.h>
+#include <mbedtls/error.h>
+#include <mbedtls/debug.h>
+#include <mbedtls/platform.h>
+#include <mbedtls/ssl_cache.h>
+// clang-format on
+
+#include "tls_e2e_t.h"
+#include "../common/utility.h"
+
+extern "C"
+{
+    int launch_tls_client(
+        struct tls_control_args* config,
+        char* server_name,
+        char* server_port);
+    int setup_tls_server(struct tls_control_args* config, char* server_port);
+};
+
+struct tls_control_args g_control_config;
+
+// This is the identity validation callback. An TLS connecting party (client or
+// server) can verify the passed in "identity" information to decide whether to
+// accept an connection reqest
+oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+
+    OE_TRACE_INFO("enclave_identity_verifier is called with parsed report:\n");
+    if (g_control_config.fail_enclave_identity_verifier_callback)
+        goto done;
+
+    // Check the enclave's security version
+    if (identity->security_version < 1)
+    {
+        OE_TRACE_ERROR(
+            "identity->security_version checking failed (%d)\n",
+            identity->security_version);
+        goto done;
+    }
+
+    // Dump an enclave's unique ID, signer ID and Product ID. They are
+    // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves In a real scenario,
+    // custom id checking should be done here
+    OE_TRACE_INFO("\nidentity->signer_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
```

Comment:
This is incorrect: You should print the unique_id i.e. MRENCLAVE
identity->signer_id[i] should be identity->unique_id[i]

Reply:
fixed